src/test/java/de/tum/in/www1/artemis/ArchitectureTest.java

```diff
@@ -160,14 +181,24 @@ public void check(JavaMethod item, ConditionEvents events) {
         };
     }
 
-    /**
-     * Checks that no class directly calls Git.commit(), but instead uses GitService.commit()
-     * This is necessary to ensure that committing is identical for all setups, with and without commit signing
-     */
-    @Test
-    void testNoDirectGitCommitCalls() {
-        ArchRule usage = noClasses().should().callMethod(Git.class, "commit").because("You should use GitService.commit() instead");
-        var classesWithoutGitService = allClasses.that(not(assignableTo(GitService.class)));
-        usage.check(classesWithoutGitService);
+    private ArchCondition<JavaMethod> haveAllParametersAnnotatedWith(DescribedPredicate<? super JavaAnnotation<?>> annotationPredicate, DescribedPredicate<JavaClass> exception) {
```

Comment:
Can you please imrove the name? Right now it reads, without checking the signature, that they can be annotated with one of them.
Maybe `haveAllParametersAnnoatedWithUnless` or something like that?